crates/wash-cli/src/run.rs

```diff
@@ -0,0 +1,266 @@
+}
+
+#[derive(Parser, Debug, Clone)]
+pub struct NetworkOptions {
+    /// Allow name to IP resolution.
+    #[clap(long = "allow-net-name-resolution")]
+    pub allow_name_resolution: bool,
+
+    /// Allow the component to connect to any peers.
+    #[clap(long = "allow-net-connect-all")]
+    pub allow_connect_all: bool,
+
+    /// Allow the component to connect to peers.
+    ///
+    /// The format expected is `<ip_addr>[:<port>[-<upper_port>]]`, where:
+    /// - `<ip_addr>` is the peer address, it can be IPv4 or IPv6
+    /// - `<port>` is the peer port we can connect to
+    /// - `<upper_port>` authorize the connection to ports ranging from `<port>` to `<upper_port>` (inclusive)
+    ///
+    /// NOTE: In future versions, we may support hostname in place of `<ip_addr>`.
+    #[clap(long = "allow-net-connect")]
+    pub allow_connect: Vec<String>,
+
+    /// Allow the component to connect to any TCP peers.
+    #[clap(long = "allow-net-connect-tcp-all")]
+    pub allow_connect_tcp_all: bool,
+
+    /// Allow the component to connect to a specific TCP peer.
+    ///
+    /// The format expected is the same as expected of `--allow-net-connect`.
+    #[clap(long = "allow-net-connect-tcp")]
+    pub allow_connect_tcp: Vec<String>,
+
+    /// Allow the component to connect to any UDP peers.
+    #[clap(long = "allow-net-connect-udp-all")]
+    pub allow_connect_udp_all: bool,
+
+    /// Allow the component to connect to UDP peers in the specified CIDR prefixes.
+    ///
+    /// The format expected is the same as expected of `--allow-net-connect`.
+    #[clap(long = "allow-net-connect-udp")]
+    pub allow_connect_udp: Vec<String>,
+
+    /// Allow the component to bind to any interface, port and protocol.
+    #[clap(long = "allow-net-bind-all")]
+    pub allow_bind_all: bool,
+
+    /// Allow the component to bind to a specific interface, specific port ranges but any protocol.
+    ///
+    /// The format expected is the same as expected of `--allow-net-connect`.
+    #[clap(long = "allow-net-bind")]
+    pub allow_bind: Vec<String>,
+
+    /// Allow the component to bind to any interface and port using the TCP protocol.
+    #[clap(long = "allow-net-bind-tcp-all")]
+    pub allow_bind_tcp_all: bool,
+
+    /// Allow the component to bind to a specific interface and specific port ranges using the TCP protocol.
+    ///
+    /// The format expected is the same as expected of `--allow-net-connect`.
+    #[clap(long = "allow-net-bind-tcp")]
+    pub allow_bind_tcp: Vec<String>,
+
+    /// Allow the component to bind to any interface and port using the UDP protocol.
+    #[clap(long = "allow-net-bind-udp-all")]
+    pub allow_bind_udp_all: bool,
+
+    /// Allow the component to bind to a specific interface and specific port ranges using the UDP protocol.
+    ///
+    /// The format expected is the same as expected of `--allow-net-connect`.
+    #[clap(long = "allow-net-bind-udp")]
+    pub allow_bind_udp: Vec<String>,
+}
```

Comment:
My general feedback is that this looks like a very robust set of options, and that might be a little overwhelming on the command line and in the code. What do you think about cutting some of these down to an initial set of flags that we can prove out the MVP, and then adding from there?

For context, I think about the use case of being able to run `wash run ./component` to be pretty constrained to components that are running with the `wasi:cli` world to start but then it can grow over time, but I still want to reconcile that with #1594 🤔 Do you have any thoughts on that?

Reply:
We may cut it down to:
-  `--allow-connect`: Allow the component to create client socket to any peers.
- `--allow-bind-all`: Allow the component to bind to any port with any protocol.
- `--allow-bind <range>`: Allow the component to bind to a range of port.

WDYT?

About reconciling it with `wash dev`, I don't have enough experience with the linking process of wasmCloud but would it means implementing every interface as host funcs or running providers locally?

Reply:
For your use case are you using `wasi-sockets` directly, or `wasi-http`? I wanted to check since, with wasmCloud, we mostly use the HTTP implementation for networking, so my thought around `wash dev` was more to get to something that after you `wash run ./component.wasm` it could also run in wasmCloud, if that makes sense 🙂

Reply:
So my idea is to allow tools that are more "POSIX-y" to use sockets and modern tools that don't need to do low level stuff should use HTTP indeed.